lib/logflare/backends/adaptor/postgres/repo.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Logflare.Backends.Adaptor.Postgres.Repo do
+  alias Logflare.Backends.SourceBackend
+  alias Logflare.Backends.Adaptor.Postgres.Supervisor
+
+  require Logger
+
+  @ast (quote do
+          use Ecto.Repo,
+            otp_app: :logflare,
+            adapter: Ecto.Adapters.Postgres
+        end)
+
+  @spec new_repository_for_source_backend(SourceBackend.t()) :: tuple()
+  def new_repository_for_source_backend(source_backend) do
+    name = Module.concat([Logflare.Repo.Postgres, "Adaptor#{source_backend.source_id}"])
+
+    case Code.ensure_compiled(name) do
+      {:module, _} -> nil
+      _ -> {:module, _, _, _} = Module.create(name, @ast, Macro.Env.location(__ENV__))
+    end
+
+    name
+  end
+
+  @spec connect_to_source_backend(Ecto.Repo.t(), SourceBackend.t(), Keyword.t()) :: :ok
+  def connect_to_source_backend(repository_module, %SourceBackend{config: config}, opts \\ []) do
+    unless Process.whereis(repository_module) do
+      %{url: url} = config
+      opts = [{:url, url} | opts]
+
+      {:ok, _} = DynamicSupervisor.start_child(Supervisor, repository_module.child_spec(opts))
+    end
+
+    :ok
+  end
+
+  @migrations [
```

Comment:
we're following the same ideas from Realtime here, we should have a properly defined schema for the postgres BE as to properly upgrade the schema if we see a need for it.

cc @chasers

Reply:
Did the change to have a migration schema so we can update schemas on psql backends if need be plus we now set a different table name based on the source.

As for the data schema changes, there's no good way of creating dynamic migrations so that will be trickier and an something that we should tackle later